Remove quicklz related codes.

zhangwenchao-123 opened the pull request:
We decide not to support quicklz compress type anymore.

Authored-by: Zhang Wenchao zhangwenchao@apache.org

<!-- Thank you for your contribution to Apache Cloudberry (Incubating)! -->

Fixes #ISSUE_Number

### What does this PR do?
<!-- Brief overview of the changes, including any major features or fixes -->

### Type of Change
- [ ] Bug fix (non-breaking change)
- [ ] New feature (non-breaking change)
- [ ] Breaking change (fix or feature with breaking changes)
- [ ] Documentation update

### Breaking Changes
<!-- Remove if not applicable. If yes, explain impact and migration path -->

### Test Plan
<!-- How did you test these changes? -->
- [ ] Unit tests added/updated
- [ ] Integration tests added/updated
- [ ] Passed `make installcheck`
- [ ] Passed `make -C src/test installcheck-cbdb-parallel`

### Impact
<!-- Remove sections that don't apply -->
**Performance:**
<!-- Any performance implications? -->

**User-facing changes:**
<!-- Any changes visible to users? -->

**Dependencies:**
<!-- New dependencies or version changes? -->

### Checklist
- [ ] Followed [contribution guide](https://cloudberry.apache.org/contribute/code)
- [ ] Added/updated documentation
- [ ] Reviewed code for security implications
- [ ] Requested review from [cloudberry committers](https://github.com/orgs/apache/teams/cloudberry-committers)

### Additional Context
<!-- Any other information that would help reviewers? Remove if none -->

### CI Skip Instructions
<!--
To skip CI builds, add the appropriate CI skip identifier to your PR title.
The identifier must:
- Be in square brackets []
- Include the word "ci" and either "skip" or "no"
- Only use for documentation-only changes or when absolutely necessary
-->

---
<!-- Join our community:
- Mailing list: [dev@cloudberry.apache.org](https://lists.apache.org/list.html?dev@cloudberry.apache.org) (subscribe: dev-subscribe@cloudberry.apache.org)
- Discussions: https://github.com/apache/cloudberry/discussions -->

tuhaihe commented:
> We decide not to support quicklz compress type anymore.

Hi @zhangwenchao-123, I'm confused about the word `we`. Is there a public thread tracking the decision process?

zhangwenchao-123 commented:
> ## Comparison with Greenplum's QuickLZ Removal Approach
> I reviewed how Greenplum handled QuickLZ removal and noticed they took a different, more gradual approach that might be worth considering:
> 
> ### Greenplum's Two-Phase Approach
> 1. **[First commit](https://github.com/greenplum-db/gpdb-archive/commit/363887e4d6abb494956bfb409acc345b0afb4d68)**: Added `gp_quicklz_fallback`
>    GUC for backward compatibility
>    
>    * Existing quicklz tables continue working
>    * New quicklz specifications automatically convert to zstd
>    * Provides graceful degradation path
> 2. **[Second commit](https://github.com/greenplum-db/gpdb-archive/commit/b2ddb8620cde72fe379604c5c6d8b2b33e02a276)**: Removed quicklz from build
>    system
>    
>    * Eliminated build-time support while keeping runtime fallback
> 
> ### This PR's Approach
> * Complete removal without fallback mechanism
> * Cleaner codebase but higher risk of breaking existing installations
> 
> ### Considerations
> **Pros of current approach:**
> 
> * Cleaner code, no technical debt
> * Clear break from unsupported feature
> 
> **Potential concerns:**
> 
> * Existing tables with quicklz compression may fail on upgrade
> * No migration path for users
> * Could cause immediate failures without clear guidance
> 
> ### Suggestions
> 1. **If keeping current approach**: Add clear error messages pointing to migration docs
>    ```c
>    ERROR: QuickLZ compression is no longer supported.
>    Please use 'zstd' or 'zlib' compression instead.
>    For migration guidance, see: [documentation link]
>    ```
> 2. Alternative: Consider implementing a temporary fallback mechanism similar to Greenplum's approach for smoother user transition
> 
> What's the team's preference on backward compatibility vs. clean removal for this change?



> ## Comparison with Greenplum's QuickLZ Removal Approach
> I reviewed how Greenplum handled QuickLZ removal and noticed they took a different, more gradual approach that might be worth considering:
> 
> ### Greenplum's Two-Phase Approach
> 1. **[First commit](https://github.com/greenplum-db/gpdb-archive/commit/363887e4d6abb494956bfb409acc345b0afb4d68)**: Added `gp_quicklz_fallback`
>    GUC for backward compatibility
>    
>    * Existing quicklz tables continue working
>    * New quicklz specifications automatically convert to zstd
>    * Provides graceful degradation path
> 2. **[Second commit](https://github.com/greenplum-db/gpdb-archive/commit/b2ddb8620cde72fe379604c5c6d8b2b33e02a276)**: Removed quicklz from build
>    system
>    
>    * Eliminated build-time support while keeping runtime fallback
> 
> ### This PR's Approach
> * Complete removal without fallback mechanism
> * Cleaner codebase but higher risk of breaking existing installations
> 
> ### Considerations
> **Pros of current approach:**
> 
> * Cleaner code, no technical debt
> * Clear break from unsupported feature
> 
> **Potential concerns:**
> 
> * Existing tables with quicklz compression may fail on upgrade
> * No migration path for users
> * Could cause immediate failures without clear guidance
> 
> ### Suggestions
> 1. **If keeping current approach**: Add clear error messages pointing to migration docs
>    ```c
>    ERROR: QuickLZ compression is no longer supported.
>    Please use 'zstd' or 'zlib' compression instead.
>    For migration guidance, see: [documentation link]
>    ```
> 2. Alternative: Consider implementing a temporary fallback mechanism similar to Greenplum's approach for smoother user transition
> 
> What's the team's preference on backward compatibility vs. clean removal for this change?

zhangwenchao-123 commented:
> ## Comparison with Greenplum's QuickLZ Removal Approach
> I reviewed how Greenplum handled QuickLZ removal and noticed they took a different, more gradual approach that might be worth considering:
> 
> ### Greenplum's Two-Phase Approach
> 1. **[First commit](https://github.com/greenplum-db/gpdb-archive/commit/363887e4d6abb494956bfb409acc345b0afb4d68)**: Added `gp_quicklz_fallback`
>    GUC for backward compatibility
>    
>    * Existing quicklz tables continue working
>    * New quicklz specifications automatically convert to zstd
>    * Provides graceful degradation path
> 2. **[Second commit](https://github.com/greenplum-db/gpdb-archive/commit/b2ddb8620cde72fe379604c5c6d8b2b33e02a276)**: Removed quicklz from build
>    system
>    
>    * Eliminated build-time support while keeping runtime fallback
> 
> ### This PR's Approach
> * Complete removal without fallback mechanism
> * Cleaner codebase but higher risk of breaking existing installations
> 
> ### Considerations
> **Pros of current approach:**
> 
> * Cleaner code, no technical debt
> * Clear break from unsupported feature
> 
> **Potential concerns:**
> 
> * Existing tables with quicklz compression may fail on upgrade
> * No migration path for users
> * Could cause immediate failures without clear guidance
> 
> ### Suggestions
> 1. **If keeping current approach**: Add clear error messages pointing to migration docs
>    ```c
>    ERROR: QuickLZ compression is no longer supported.
>    Please use 'zstd' or 'zlib' compression instead.
>    For migration guidance, see: [documentation link]
>    ```
> 2. Alternative: Consider implementing a temporary fallback mechanism similar to Greenplum's approach for smoother user transition
> 
> What's the team's preference on backward compatibility vs. clean removal for this change?

The reason why we do this is as the quicklz has license argument, so it's better to remove all of the codes which have `quicklz` key words. As we have changed some files such as CI related files in our repo, it can't cherry-pick from gp upstream directly, this approach may be better. For migration, we need time to discuss how to achieve it, but maybe we should remove quicklz firstly to avoid the danger of unnecessary license argument.

tuhaihe commented:
> > What's the team's preference on backward compatibility vs. clean removal for this change?
> 
> The reason why we do this is as the quicklz has license argument, so it's better to remove all of the codes which have `quicklz` key words. As we have changed some files such as CI related files in our repo, it can't cherry-pick from gp upstream directly, this approach may be better. For migration, we need time to discuss how to achieve it, but maybe we should remove quicklz firstly to avoid the danger of unnecessary license argument.

Hi @zhangwenchao-123, in my opinion, including the keyword `quicklz` in the source code is okay. Also, some files related to the old CI have been removed from the Cloudberry repo, which doesn't prevent us from cherry-picking the relevant commits from GP to Cloudberry.

I think we should adopt the GP approach and cherry-pick the relevant commits to Cloudberry. (If that's not enough, we need to improve based on these commits.) This also helps keep the GP method at its best for traditional GP users.

my-ship-it commented:
> ```c
> ERROR: QuickLZ compression is no longer supported.
> Please use 'zstd' or 'zlib' compression instead.
> For migration guidance, see: [documentation link]
> ```

Thanks Ed.

Since Cloudberry 2.0 is a new version, I personally prefer to completely remove quicklz support and provide an clear error message.

zhangwenchao-123 commented:
> > ```c
> > ERROR: QuickLZ compression is no longer supported.
> > Please use 'zstd' or 'zlib' compression instead.
> > For migration guidance, see: [documentation link]
> > ```
> 
> Thanks Ed.
> 
> Since Cloudberry 2.0 is a new major version, I personally prefer to completely remove quicklz support and provide an clear error message.
Agree with me and pushed new commit.
